KB-10098 | BE|DEV|API to Provide the connection status between two users.

tarentomaheshvakkund opened the pull request.

tarentomaheshvakkund committed:
KB-10098 | BE|DEV|API to Provide the connection status between two users.

1.API added which takes two user id and check the connection status and returns the response.

tarentomaheshvakkund committed:
Constants added

karthik-tarento commented on src/main/java/org/sunbird/cb/hubservices/network/controller/ConnectionProfileController.java:
```diff
@@ -71,4 +73,10 @@ public ResponseEntity<Response> findEstablished(@RequestHeader(required = false)
+	@PostMapping(Constants.RELATIONSHIP)
+	public ResponseEntity<Response> getRelationshipBetweenUsers( @RequestBody Map<String, Object> requestBody) {
+		Response response=profileService.getRelationshipBetweenUsers(requestBody);
+		return new ResponseEntity<>(response,HttpStatus.OK);
```
We shouldn't send 200 OK by default... can you enhance the code to get the proper code from response object?

tarentomaheshvakkund committed:
Code Review Comments addressed.

1. Added the authentication code for auth token validation.
2. Followed the Sunbird Response structure for returning response.
3. Handled logger.

karthik-tarento commented on src/main/java/org/sunbird/cb/hubservices/serviceimpl/ProfileService.java:
```diff
@@ -82,4 +90,27 @@ public Response multiSearchProfiles(String userId, MultiSearch mSearchRequest, S
+			Map<String, String> map = connectionService.getRelationshipBetweenUsers(fromUserId,
+					toUserId);
+			response.getResult().put("response",map);
+			response.getParams().setStatus(Constants.OK);
+			response.setResponseCode(HttpStatus.OK);
```
this is already done in createDefaultResponse method -- let's keep that value...
